homeassistant/components/cloud/stt.py

```diff
@@ -13,37 +13,37 @@
     AudioCodecs,
     AudioFormats,
     AudioSampleRates,
-    Provider,
     SpeechMetadata,
     SpeechResult,
     SpeechResultState,
+    SpeechToTextEntity,
 )
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 from .client import CloudClient
 from .const import DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
 
-async def async_get_engine(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
-    discovery_info: DiscoveryInfoType | None = None,
-) -> CloudProvider:
-    """Set up Cloud speech component."""
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Demo speech platform via config entry."""
     cloud: Cloud[CloudClient] = hass.data[DOMAIN]
+    async_add_entities([CloudProviderEntity(cloud)])
 
-    cloud_provider = CloudProvider(cloud)
-    if discovery_info is not None:
-        discovery_info["platform_loaded"].set()
-    return cloud_provider
 
-
-class CloudProvider(Provider):
+class CloudProviderEntity(SpeechToTextEntity):
```

Comment:
I'll follow up with a PR that syncs the entity being available with cloud stop/start callbacks which corresponds to cloud login/logout.